src/response_sender.h

```diff
@@ -46,7 +46,16 @@ class ResponseSender {
   intptr_t request_address_;
   intptr_t response_factory_address_;
   std::unique_ptr<SharedMemoryManager>& shm_pool_;
+  // The flag to indicate if the response sender is closed. It is set to true
+  // once the TRITONSERVER_RESPONSE_COMPLETE_FINAL flag is set, meaning that the
+  // response_sender should not be used anymore. This flag is separate from the
+  // `is_response_factory_cleaned_` flag because errors might occur after
```

Comment:
Can you elaborate on the errors that might occur after sending complete_final flag but before deleting the response_factory? Is it the model that is holding on to the response_sender forever?

Reply:
When the complete final flag is set, it is still possible that some error occurs before the response actually gets sent to the backend. For example, [here](https://github.com/triton-inference-server/python_backend/blob/main/src/response_sender.cc#L78) and [here](https://github.com/triton-inference-server/python_backend/blob/main/src/response_sender.cc#L83).
In this case, the `response_factory` will not be automatically deleted in the backend, and can only be deleted within the destructor of `response_sender` by sending the clean up message to the backend.

I think this bool variable `closed_` is just to make sure users are not using the response sender after they set the final flag, It can not be used to determine if the response_factory is cleaned up from the backend side or not.